### ISSUE TITLE
Provide a script to generate Git for Windows installers

### ISSUE THREAD

**Issue by dscho** [git-for-windows/git#13]
Most likely, a lot of https://github.com/msysgit/msysgit/tree/master/share/WinGit can be scavenged to make this happen.

**Comment by dscho** [comment]
This work has been done by @sschuberth, merged in https://github.com/git-for-windows/MSYS2-packages/pull/7

**Comment by sschuberth** [comment]
@dscho My PR you're referring to is for an *SDK* installer. However, the title of this issue sounds as if it was about an end-user installer for the MSYS2-based Git for Windows. Then again, this issue is labelled "sdk". Can you please clarify by either adjust the title or the label of this issue?

**Comment by dscho** [comment]
Whoops, I misread this ticket.

**Comment by lygstate** [comment]
Really need the installer,

**Comment by PhilipOakley** [comment]
From: Yonggang Luo 
  Really need the installer,

Yonggang Lou,
Which aretfact are you wanting an installer for ?
A. The run time Git for Windows for V2.0 or newer
B. The run time for the older 'Git for Windows' at V1.9.5 + maintenance releases
C. the Software Development Kit (SDK) to help develop the new release of >V2.x.x
D. the Development Environment for the older 'Git for Windows' aka MsysGit?

Option B (latest) was released a few days ago. [1]
Option D is still available. [2]
Option C is possible 

[1] https://github.com/msysgit/git/releases/tag/v1.9.5.msysgit.1 is the 
corresponding source code release. Binaries are available from 
https://github.com/msysgit/msysgit/releases/tag/Git-1.9.5-preview20150319.
[2] https://github.com/msysgit

**Comment by lygstate** [comment]
@PhilipOakley  I am looking for A

**Comment by PhilipOakley** [comment]
From: Yonggang Luo 
  @PhilipOakley I am looking for A

A. The run time Git for Windows for V2.0 or newer

I thought someone had reported that they had rebased V2.x.y in the MsysGit environment on their list https://groups.google.com/forum/?hl=en_US?hl%3Den#!forum/msysgit but a quick search of my notifications didn't spot it. 

You may have better look looking & asking there. 

The new MSYS2 based Git for Windows is coming along so you could help there/here...

Philip

**Comment by dscho** [comment]
> Really need the installer

So you really want something. How much (in addition to one-line comments) are you prepared to give in return?

**Comment by nalla** [comment]
For the ones who does want to try out the current state of development, and even want to help test it, here is a little *HOWTO*:

1. Download *[net-installer](https://github.com/git-for-windows/build-extra/releases)* for the *Git for Windows SDK* and install it.

2. Startup the *MinGW* *Git for Windows SDK* shell (Look for shortcuts in the *Git for Windows SDK* install folder)

3. Clone the *[build-extra](https://github.com/git-for-windows/build-extra)* repository and checkout the feature branch *[portable-release](https://github.com/git-for-windows/build-extra/tree/portable-release)* within the *Git for Windows SDK*.

4. Change in the `portable` directory of the new clone.

5. Run the `./release.sh <version-no>` shell script. That *should* create you a portable installer for *Git for Windows* in your home directory.

6. Test it!

**Comment by weakcamel** [comment]
Hello,
I'd like to look around the repos and see if there is a small piece where I could help.
Perhaps the Perl upgrade, but that's too early to tell :-)
However, I just cloned the `git-for-windows-sdk` and the `build-extra` and I fail to see the portable-release branch anywhere.
What am I doing wrong?

**Comment by dscho** [comment]
> Perhaps the Perl upgrade, but that's too early to tell :-)

The nice thing is that with our switch to MSys2, we have already pretty much up-to-date software packages. Perl, for example, is reported to be at version:

```
$ pacman -Q perl
perl 5.20.2-1
```

> However, I just cloned the git-for-windows-sdk and the build-extra and I fail to see the portable-release branch anywhere.

The best way to get started is to use the *new* SDK, as installed by the new net installer: https://github.com/git-for-windows/build-extra/releases/tag/net-installer-1.0.0-rc-1

As to the `portable-release` branch: it was merged (see https://github.com/git-for-windows/build-extra/commit/764430ff182f203c3e820413a2ca6b6e2a365b48).

Now, if you look for rough edges to even out, I would be really glad to have you on board! It would be real nice, for example, if you would have a look at [this ticket](https://github.com/git-for-windows/git/issues/52); A relatively easy task would be to squash the compile warnings mentioned by @PhilipOakley, a more involved task would be the performance issues with `git status` or the permission problems in the `filesystem` package. (I fixed the `git gui` problem mentioned in that ticket and am currently cleaning up those patches).
